specs/_features/eip7594/das-core.md

```diff
@@ -239,7 +239,27 @@ To custody a particular column, a node joins the respective gossip subnet. Verif
 
 ## Peer sampling
 
-A node SHOULD maintain a diverse set of peers for each column and each slot by verifying responsiveness to sample queries. At each slot, a node makes `SAMPLES_PER_SLOT` queries for samples from their peers via `DataColumnSidecarsByRoot` request. A node utilizes `get_custody_columns` helper to determine which peer(s) to request from. If a node has enough good/honest peers across all rows and columns, this has a high chance of success.
+### Sample selection
+
+At each slot, a node SHOULD select at least `SAMPLES_PER_SLOT` column IDs for sampling. It is recommended to use uniform random selection without replacement based on local randomness. Sampling is considered successful if the node manages to retrieve all selected columns.
+
+Alternatively, a node MAY use LossyDAS selecting more than `SAMPLES_PER_SLOT` columns while allowing some missing, respecting the same target false positive threshold (the probability of successful sampling of an unavailable block) as dictated by `SAMPLES_PER_SLOT`. The table below shows the number of samples and the number of allowed missing columns for this threshold.  
+
```

Comment:
I have such a function in the backing Jupyter notebook using the CDF and the PPF, although I think it not the most elegant version. I will try to add it as a function as tou suggest.

Reply:
I have added the `get_extended_sample_count(samples_per_slot, allowed_failures) -> sample_count` helper function.

At this point, we might also remove the table, or move it to the FAQ section (not sure which one is better).

Reply:
As long as the values match the function, I think we should keep the table! It gives the reader a good idea of how many missing samples are allowed. I like the function, but I don't love the dependency on `scipy` 😅

Reply:
OK, so the table stays.
As for scipy: if really needed, I could try to eliminate scipy by manually adding the formula ..... let me know.

Reply:
alternative is
```
from math import comb

def hyperg_cdf(k, M, n, N):
  return sum([comb(n,i)*comb(M-n,N-i)/comb(M,N) for i in range(k+1)])
```
If this sounds better, I can change it. I was trying to search for import examples, but couldn't find how it should be handled.

Reply:
yeah, I think importing `math` is better than `scipy` for spec readers.

Reply:
turns out `math.comb` has the wrong definition in Python, failing on negative `k` instead of returning 0 as for `k>n`. Can be handled, but I've preferred checking in a self-contained version without imports.

Reply:
Another point: should we have `allowed_failures` as a global parameter or it can be configured at each node?

If it can be configured at each node, is there a way to figure out what it should be?

Reply:
> Why do we have to bother putting the number of samples per slot as a parameter of the function? Is there any case that you will use the number other than SAMPLES_PER_SLOT?
> How about get_extended_sample_count(allowed_failures) instead?
Right. On the other hand, I'm not a big fan of functions that silently use constants, but there is also NUMBER_OF_COLUMNS, which is currently silently used inside. So there are actually one more options:
- get_extended_sample_count(number_of_columns, samples_per_slot, allowed_failures)

I'm fine with any of these. Happy to modify it to whichever aligns most with current practices.

Reply:
> Another point: should we have allowed_failures as a global parameter or it can be configured at each node?

It can be a local decision. It is a local optimization, which can also be part of an incremental strategy.

> If it can be configured at each node, is there a way to figure out what it should be?

It depends on expectations, which can depend on implementation details, on peer quality, on past sampling performance etc. It can also be used as part of an incremental strategy.
I don't think we can give a recommendation at the EIP level on what to choose. This is more about not mandating an "all out of SMAPLES_PER_SLOT", and about clarifying what are the alternatives.

Reply:
So, in my opinion, it should be either
`get_extended_sample_count(number_of_columns, samples_per_slot, allowed_failures)` or
`get_extended_sample_count(allowed_failures)`

However, I prefer the latter. I think there is no reason for `get_extended_sample_count(samples_per_slot, allowed_failures)`.

Reply:
I've changed it to `get_extended_sample_count(allowed_failures)`